extmod/uasyncio/funcs.py

```diff
@@ -123,7 +123,7 @@ def done(t, er):
 
     # Either this gather was cancelled, or one of the sub-tasks raised an exception with
     # return_exceptions==False, so reraise the exception here.
-    if state is not 0:
+    if state != 0:
```

Comment:
`state` can be an integer or an exception type, so that's why `is not` is used

Reply:
`is not 0` only works because of the implementation detail of MP_SMALL_INT but in general `is` should not be used with ints.

Reply:
Yes you are right.  So I'll need to check in detail what happens for `state != 0`  when state is an exception instance.

Reply:
Won't `if state:` work since 0 is falsy but any exception will be truthy?

Reply:
Yes that's a good idea.  At this point `state` is either `0` or an exception instance (because in the latter case it's raising it).